pkg/ccl/sqlproxyccl/proxy_handler_test.go

```diff
@@ -657,6 +657,24 @@ func TestProxyRefuseConn(t *testing.T) {
 	require.Equal(t, int64(0), s.metrics.AuthFailedCount.Count())
 }
 
+func TestProxyTCPProbe(t *testing.T) {
```

Comment:
I could be missing something, but I'm not sure what this test is asserting actually. I think this test would likely pass even without these changes. Is there a way we can assert on how the `handle` method treats the connection closing without sending anything? If not, that's fine, but I don't see what this test does except check that we can accept TCP connections, but the other tests do that already.